File: nup/src/test/java/FakeTaskRunner.java
// Copyright 2016 Daniel Erat <dan@erat.org>
// All rights reserved.

package org.erat.nup.test;

import org.erat.nup.TaskRunner;

/** Fake implemenation of TaskRunner that runs tasks immediately. */
public class FakeTaskRunner extends TaskRunner {
    public void runInBackground(final Runnable runnable) {
        // Use a new thread to avoid triggering running-on-non-UI-thread asserts.
        Thread thread = new Thread(runnable);
        thread.start();
        try {
            thread.join();
        } catch (InterruptedException e) {
            throw new IllegalStateException(e);
        }
    }
}

File: nup/src/test/java/FakeTaskRunner.kt
// Copyright 2016 Daniel Erat <dan@erat.org>
// All rights reserved.
package org.erat.nup.test

import org.erat.nup.TaskRunner

/** Fake implemenation of TaskRunner that runs tasks immediately.  */
class FakeTaskRunner : TaskRunner() {
    override fun runInBackground(runnable: Runnable) {
        // Use a new thread to avoid triggering running-on-non-UI-thread asserts.
        val thread = Thread(runnable)
        thread.start()
        try {
            thread.join()
        } catch (e: InterruptedException) {
            throw IllegalStateException(e)
        }
    }
}